### ISSUE TITLE
GET/POST : enable the cache control

### ISSUE THREAD

**Issue by elouanKeryell-Even** [gisaia/ARLAS-server#91]
In gitlab by @sylvaingaudan on Aug 11, 2017, 08:09

Every `GET`/`POST` service used for retrieving data (read only services) should implement the cache control that is already referenced in the API:
```java
@ApiParam(value = "max-age-cache", required = false)
@QueryParam(value = "max-age-cache") Integer maxagecache
```
This is done by setting the cache control on the response, if provided:
```java
Reponse reponse = Response.ok(...);
if(DEFAULT_MAX_AGE_CACHE!=null || maxagecache!=null){
   if(maxagecache == null){
      maxagecache=DEFAULT_MAX_AGE_CACHE;// DEFAULT_MAX_AGE_CACHE defined in ARLAS configuration file
   }
   CacheControl cc = new CacheControl();
   cc.setPrivate(false);
   cc.setNoCache(false);
   cc.setNoTransform(true);
   cc.setMaxAge(maxagecache);
   reponse.cacheControl(cc);
}
return reponse.build();
```

**Comment by elouanKeryell-Even** [comment]
In gitlab by @sylvaingaudan on Aug 11, 2017, 08:10

added ~2424197 label

**Comment by elouanKeryell-Even** [comment]
In gitlab by @sfalquier on Aug 18, 2017, 09:11

added ~2424198 label

**Comment by elouanKeryell-Even** [comment]
In gitlab by @sfalquier on Aug 18, 2017, 14:25

added ~2424199 and removed ~2424198 labels

**Comment by elouanKeryell-Even** [comment]
In gitlab by @sfalquier on Aug 18, 2017, 17:49

mentioned in commit a2bcbb0aecffad88173d5768b7355bb8ba603b4c

**Comment by elouanKeryell-Even** [comment]
In gitlab by @sfalquier on Aug 18, 2017, 17:52

assigned to @sfalquier

**Comment by elouanKeryell-Even** [comment]
In gitlab by @sfalquier on Aug 18, 2017, 17:58

As far as I understand, CacheControl adds cache related http headers to response that enable the http client to manage its cache on its side. As a result, it's not a server side cache and it fully relies on client implementation both on POST or GET request.

@sylvaingaudan : Even if it's a nice hack, is it what you are expecting? I'd bet that you are looking for a server-side cache system.

**Comment by elouanKeryell-Even** [comment]
In gitlab by @sfalquier on Aug 18, 2017, 19:05

mentioned in commit 551a3aec0be7e6c732955efa8aeb4964a67dfdf6

**Comment by elouanKeryell-Even** [comment]
In gitlab by @sylvaingaudan on Aug 18, 2017, 22:39

The goal is to have a caché contrôled by the client and managed by the server/browser : 
- the client tells how long is the cache
- the server response is in accordance with the requedt. 
- the browser avoids calls based on the cache.

**Comment by elouanKeryell-Even** [comment]
In gitlab by @sfalquier on Aug 21, 2017, 09:05

Ok, so there is no need of server-side cache.

But why the browser/client tells how long should last the cache?

It looks like a strange behavior that a client would be able to make several similar calls with different max-age-cache param making the server doing the same ES requests several times since max-age-cache is part of the api request.
On top of that, I found it more reliable if cache control headers are managed on server-side with conf (maybe with different cache control policies if needed according to api needs). I don't understand why a client would tell Arlas to apply cache control headers on responses by giving it the TTL as request param.

**Comment by elouanKeryell-Even** [comment]
In gitlab by @sfalquier on Aug 25, 2017, 09:02

mentioned in merge request !83

**Comment by elouanKeryell-Even** [comment]
In gitlab by @MohamedHamouGisaia on Aug 25, 2017, 09:32

closed via merge request !83

**Comment by elouanKeryell-Even** [comment]
In gitlab by @MohamedHamouGisaia on Aug 25, 2017, 09:32

closed via commit 5002482b4db38c533f13d756949d52baa5a91952

**Comment by elouanKeryell-Even** [comment]
In gitlab by @MohamedHamouGisaia on Aug 25, 2017, 11:48

mentioned in commit 5002482b4db38c533f13d756949d52baa5a91952
